executor/aggregate.go

```diff
@@ -355,15 +354,16 @@ func (w *HashAggPartialWorker) run(ctx sessionctx.Context, waitGroup *sync.WaitG
 }
 
 func (w *HashAggPartialWorker) updatePartialResult(ctx sessionctx.Context, sc *stmtctx.StatementContext, chk *chunk.Chunk, finalConcurrency int) (err error) {
-	inputIter := chunk.NewIterator4Chunk(chk)
-	for row := inputIter.Begin(); row != inputIter.End(); row = inputIter.Next() {
-		groupKey, err := w.getGroupKey(sc, row)
-		if err != nil {
-			return err
-		}
-		partialResults := w.getPartialResult(sc, groupKey, w.partialResultsMap)
-		for i, af := range w.aggFuncs {
-			if err = af.UpdatePartialResult(ctx, []chunk.Row{row}, partialResults[i]); err != nil {
+	err = w.getGroupKey(sc, chk)
+	if err != nil {
+		return err
+	}
+
+	partialResults := w.getPartialResult(sc, w.groupKey, w.partialResultsMap)
+	numRows := chk.NumRows()
+	for i := 0; i < numRows; i++ {
+		for j, af := range w.aggFuncs {
+			if err = af.UpdatePartialResult(ctx, []chunk.Row{chk.GetRow(i)}, partialResults[i][j]); err != nil {
```

Comment:
We'll further make the `UpdatePartialResult` be vectorized?

Reply:
Sure, the ```UpdatePartialResult``` is related to the ```aggregate functions```. We will support ```aggregate functions``` to update the results in a vectorized way in the next PR.